api/src/test/java/org/pmiops/workbench/db/service/UserRecentResourceServiceTest.java

```diff
@@ -0,0 +1,147 @@
+public class UserRecentResourceServiceTest {
+
+  UserRecentResourceServiceImpl userRecentResourceService;
+
+  @Autowired
+  UserDao userDao;
+  @Autowired
+  WorkspaceDao workspaceDao;
+  @Autowired
+  CohortDao cohortDao;
+  @Autowired
+  UserRecentResourceDao notebookCohortCacheDao;
+  @Mock
+  WorkbenchConfig config;
+
+  private User newUser = new User();
+  private Workspace newWorkspace = new Workspace();
+  private Long cohortId;
+  private long workspaceId = 1l;
+  private long userId = 1l;
+
+  @Before
+  public void setUp() {
+    newUser.setUserId(userId);
+    userDao.save(newUser);
+    newWorkspace.setWorkspaceId(workspaceId);
+    workspaceDao.save(newWorkspace);
+    Cohort cohort = new Cohort();
+    cohort.setWorkspaceId(workspaceId);
+    cohortId = cohortDao.save(cohort).getCohortId();
+    config.userRecentResourceConfig = new WorkbenchConfig.UserRecentResourceConfig();
+    config.userRecentResourceConfig.userEntrycount = 3;
+    userRecentResourceService = new UserRecentResourceServiceImpl(Providers.of(config));
+    userRecentResourceService.setDao(notebookCohortCacheDao);
+  }
+
+
+  @Test
+  public void testInsertCohortEntry() {
+    userRecentResourceService.updateCohortEntry(workspaceId, userId, cohortId, new Timestamp(new Date().getTime()));
+    long noOfRows = userRecentResourceService.getDao().count();
+    assertEquals(noOfRows, 1);
+    Cohort cohort = new Cohort();
+    cohort.setWorkspaceId(1l);
+    cohortId = cohortDao.save(cohort).getCohortId();
+    userRecentResourceService.updateCohortEntry(workspaceId, userId, cohortId, new Timestamp(new Date().getTime()));
+    noOfRows = userRecentResourceService.getDao().count();
+    assertEquals(noOfRows, 2);
+  }
+
+  @Test
+  public void testInsertNotebookEntry() {
+    userRecentResourceService.updateNotebookEntry(workspaceId, userId, "notebook1", new Timestamp(new Date().getTime()));
+    long noOfRows = userRecentResourceService.getDao().count();
+    assertEquals(noOfRows, 1);
+    userRecentResourceService.updateNotebookEntry(workspaceId, userId, "notebook2", new Timestamp(new Date().getTime()));
+    noOfRows = userRecentResourceService.getDao().count();
+    assertEquals(noOfRows, 2);
+
+  }
+
+  @Test
+  public void testUpdateCohortAccessTime() {
+    userRecentResourceService.updateCohortEntry(workspaceId, userId, cohortId, new Timestamp(new Date(2018, 3, 3).getTime()));
+    long noOfRows = userRecentResourceService.getDao().count();
+    assertEquals(noOfRows, 1);
+    userRecentResourceService.updateCohortEntry(workspaceId, userId, cohortId, new Timestamp(new Date().getTime()));
+    noOfRows = userRecentResourceService.getDao().count();
+    assertEquals(noOfRows, 1);
+  }
+
+  @Test
+  public void testUpdateNotebookAccessTime() {
+    userRecentResourceService.updateNotebookEntry(workspaceId, userId, "notebook1", new Timestamp(new Date(2018, 3, 3).getTime()));
+    long noOfRows = userRecentResourceService.getDao().count();
```

Comment:
ultranit: `noOfRows` looks a bit awkward to me, I might go with `numRows`. Alternatively I might also just inline this in the assert below